unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/WorldChatWindowHUD/NearbyMembersHUDController.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+
+namespace DCL.Chat.HUD
+{
+    public class NearbyMembersHUDController : IDisposable
+    {
+        private readonly IChannelMembersComponentView view;
+        private readonly DataStore_Player playerDataStore;
+        private readonly IUserProfileBridge userProfileBridge;
+        private bool isVisible;
+        private string currentSearchText;
+
+        public IChannelMembersComponentView View => view;
+
+        public NearbyMembersHUDController(
+            IChannelMembersComponentView view,
+            DataStore_Player playerDataStore,
+            IUserProfileBridge userProfileBridge)
+        {
+            this.view = view;
+            this.playerDataStore = playerDataStore;
+            this.userProfileBridge = userProfileBridge;
+
+            playerDataStore.otherPlayers.OnAdded += OnNearbyPlayersAdded;
+            playerDataStore.otherPlayers.OnRemoved += OnNearbyPlayersRemoved;
+            view.OnSearchUpdated += OnSearchPlayers;
+        }
+
+        public void SetVisibility(bool visible)
+        {
+            isVisible = visible;
+
+            if (visible)
+            {
+                view.Show();
+                view.ClearSearchInput(false);
+                LoadCurrentNearbyPlayers();
+            }
+            else
+                view.Hide();
+        }
+
+        public void ClearSearch() =>
+            view.ClearSearchInput();
+
+        public void Dispose()
+        {
+            playerDataStore.otherPlayers.OnAdded -= OnNearbyPlayersAdded;
+            playerDataStore.otherPlayers.OnRemoved -= OnNearbyPlayersRemoved;
+            view.OnSearchUpdated -= OnSearchPlayers;
+            view.Dispose();
+        }
+
+        private void LoadCurrentNearbyPlayers(string textFilter = "")
+        {
+            view.ClearAllEntries();
+
+            foreach (var player in playerDataStore.otherPlayers.Get())
+            {
+                if (!string.IsNullOrEmpty(textFilter) &&
+                    !player.Value.name.ToLower().Contains(textFilter.ToLower()))
```

Comment:
Just use `StringComparison.OrdinalIgnoreCase` to avoid allocations

Reply:
Done: https://github.com/decentraland/unity-renderer/pull/4679/commits/b84d563a26ef398b47c7c8e158bee54c0f0e72cf